traffic_ops/traffic_ops_golang/federation_resolvers/federation_resolvers.go

```diff
@@ -136,3 +155,101 @@ func Read(w http.ResponseWriter, r *http.Request) {
 
 	api.WriteResp(w, r, resolvers)
 }
+
+// Delete is the handler for DELETE requests to /federation_resolvers.
+func Delete(w http.ResponseWriter, r *http.Request) {
+	inf, sysErr, userErr, errCode := api.NewInfo(r, []string{"id"}, []string{"id"})
+	tx := inf.Tx.Tx
+	if sysErr != nil || userErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	alert, respObj, userErr, sysErr, statusCode := deleteFederationResolver(inf)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, statusCode, userErr, sysErr)
+		return
+	}
+
+	api.WriteRespAlertObj(w, r, tc.SuccessLevel, alert.Text, respObj)
+}
+
+func deleteFederationResolver(inf *api.APIInfo) (tc.Alert, tc.FederationResolver, error, error, int) {
+	var userErr error
+	var sysErr error
+	var statusCode = http.StatusOK
+	var alert tc.Alert
+	var result tc.FederationResolver
+
+	err := inf.Tx.Tx.QueryRow(deleteQuery, inf.IntParams["id"]).Scan(&result.ID, &result.IPAddress, &result.Type)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			userErr = fmt.Errorf("No federation resolver by ID %d", inf.IntParams["id"])
+			statusCode = http.StatusNotFound
+		} else {
+			userErr, sysErr, statusCode = api.ParseDBError(err)
+		}
+		return userErr, sysErr, statusCode, alert, result
+	}
+
+	changeLogMsg := fmt.Sprintf("FEDERATION_RESOLVER: %s, ID: %d, ACTION: Deleted", *result.IPAddress, *result.ID)
+	api.CreateChangeLogRawTx(api.ApiChange, changeLogMsg, inf.User, inf.Tx.Tx)
+
+	alertMsg := fmt.Sprintf("Federation resolver deleted [ IP = %s ] with id: %d", *result.IPAddress, *result.ID)
+	alert = tc.Alert{
+		Level: tc.SuccessLevel.String(),
+		Text:  alertMsg,
+	}
+
+	return userErr, sysErr, statusCode, alert, result
+}
+
+// DeleteByID is the handler for DELETE requests to /federation_resolvers/{{ID}}.
+func DeleteByID(w http.ResponseWriter, r *http.Request) {
+	inf, sysErr, userErr, errCode := api.NewInfo(r, []string{"id"}, []string{"id"})
+	tx := inf.Tx.Tx
+	if sysErr != nil || userErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	alert, respObj, userErr, sysErr, statusCode := deleteFederationResolver(inf)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, statusCode, userErr, sysErr)
+		return
+	}
+
+	var resp = struct {
+		tc.Alerts
+		Response *tc.FederationResolver `json:"response,omitempty"`
+	}{
+		tc.Alerts{
+			Alerts: []tc.Alert{
+				alert,
+				tc.Alert{
+					Level: tc.WarnLevel.String(),
+					Text:  "This endpoint is deprecated, please use '/federation_resolvers' instead",
```

Comment:
nit but should this be `use '/federation_resolvers?id={id}' instead `

Reply:
Well it's not *wrong*, but I can certainly add that.

Reply:
not wrong at all and completely unblocking comment so take it or leave it :)